### PR TITLE
Add a test logger

### PR DESCRIPTION
**Pull request by Kemosabert**
In this PR I added the option to use a test logger instead of the default one, this should improve the logs during the tests, grouping them with the failed test 👍

### CODE REVIEW

**Comment by kdisneur** [comment]
Task linked: [CU-d31b9b Add ZapTest to hide logs from successful tests](https://app.clickup.com/t/d31b9b)

**Comment by julien-leclercq** [comment]
Did you manually tried ?

**Comment by Kemosabert** [comment]
Ofcourse I tried ;)

Screenshot 2021-01-28 at 15 32 37

File: platform/logging/logger.go [hunk 1]
```diff
@@ -2,13 +2,22 @@ package logging
...
 )
 
 // Logger follow the standard go Logger and only give access to `Println`
-type Logger struct {
+type Logger interface {
+	Printf(string, ...interface{})
+	Println(...interface{})
+	PrintRequestResponse(RequestAttributes, TraceAttribute, DurationAttribute, RemoteAddressAttribute, string)
+	Sync() error
+}
```
Comment (julien-leclercq) [inline platform/logging/logger.go:18]: What is the point of that interface ?

File: platform/logging/logger.go [hunk 2]
```diff
@@ -34,29 +43,38 @@ func RequestAttribute(method string, path string, statusCode int) RequestAttribu
...
+
+// NewTestLogger creates a new logger meant to be used during testing
+func NewTestLogger(t *testing.T) (Logger, error) {
+	zLogger := zaptest.NewLogger(t, zaptest.WrapOptions())
 
-	return &logger, nil
+	return &logger{
+		logger: zLogger,
+	}, nil
 }
```
Comment (julien-leclercq) [inline platform/logging/logger.go:64]: I think this is the sufficient change

File: platform/logging/logger.go (hunk 1 above)
Comment (Kemosabert) [reply platform/logging/logger.go:18]: You are right, I thought the type of the logger would be different, but seems like it's fine, I will change it out

Comment (julien-leclercq) [reply platform/logging/logger.go:18]: yeah that was the whole point of using this lib :)

Comment (Kemosabert) [reply platform/logging/logger.go]: Updated it :)

**Review by julien-leclercq — APPROVED** [review]

**Pull request merged**